packages/xo-web/src/xo-app/backup-ng/new/index.js

```diff
@@ -398,28 +398,42 @@ export default [
       showScheduleModal: (
         { saveSchedule },
         storedSchedule = DEFAULT_SCHEDULE
-      ) => async ({ copyMode, exportMode, snapshotMode }) => {
+      ) => async (
+        { copyMode, exportMode, snapshotMode },
+        { intl: { formatMessage } }
+      ) => {
         const schedule = await form({
-          body: <NewSchedule modes={{ copyMode, exportMode, snapshotMode }} />,
           defaultValue: storedSchedule,
-          icon: 'schedule',
+          render: props => (
+            <NewSchedule
+              modes={{ copyMode, exportMode, snapshotMode }}
+              {...props}
+            />
+          ),
+          header: (
+            <span>
+              <Icon icon='schedule' /> {_('schedule')}
+            </span>
+          ),
           size: 'large',
-          title: _('schedule'),
+          handler: value => {
+            if (
+              !(
+                (exportMode && value.exportRetention > 0) ||
+                (copyMode && value.copyRetention > 0) ||
+                (snapshotMode && value.snapshotRetention > 0)
+              )
+            ) {
+              throw new Error(formatMessage(messages.retentionNeeded))
```

Comment:
Why do you need to internationalize the error message?

Reply:
Because is a customized error.

Reply:
You're right :+1:

Reply:
After discussing, we decided to internationalize the UI errors  and it will be better if we create a [new class of errors](https://github.com/vatesfr/xen-orchestra/issues/3597) dedicated to the users errors.